### DISCUSSION TITLE
How to open a file from project_panel in Vertical/Horizontal split in vim mode

### DISCUSSION THREAD

**Discussion by milosgajdos** [zed-industries/zed#16901] [Feature Requests]
I'm trying to figure out if there is a way to open a file from the project panel in the vim mode.

I use [NERDTree](https://github.com/preservim/nerdtree) in vim and it has these small key shortcuts (i/s) that let you open files in horozintal/vertical splits.

My brain is now so hardwired to these shortcuts that I'd like to have them in Zed, too.

One of the things I considered was doing some combo of `"project_panel::OpenPermanent"` and `"pane::SplitVertical"` or so and so. Right now I'm doing this:
1. Open a new vertical split
2. Cmd+P to open a file in it

That feels very clunky and like I said my brain is hard to rewire. Curious if there is some hack I'm missing.

**Comment by mahmudzero** [comment]
> I'm trying to figure out if there is a way to open a file from the project panel in the vim mode.
> 
> 
> 
> I use [NERDTree](https://github.com/preservim/nerdtree) in vim and it has these small key shortcuts (i/s) that let you open files in horozintal/vertical splits.
> 
> 
> 
> My brain is now so hardwired to these shortcuts that I'd like to have them in Zed, too.
> 
> 
> 
> One of the things I considered was doing some combo of `"project_panel::OpenPermanent"` and `"pane::SplitVertical"` or so and so. Right now I'm doing this:
> 
> 1. Open a new vertical split
> 
> 2. Cmd+P to open a file in it
> 
> 
> 
> That feels very clunky and like I said my brain is hard to rewire. Curious if there is some hack I'm missing.

Replying so I'm in this thread if there's ever a native way to do it

currently doing same jangle of key strokes and it's annoying

**Comment by patrykk21** [comment]
I need this

**Comment by goelankitt** [comment]
I really need this!

**Comment by goelankitt** [comment]
I am trying to do this by setting a keymap for this. But I can't get it to work.
`  
{
    "context": "ProjectPanel && not_editing",
    "bindings": {
      "o": "project_panel::Open && pane::SplitVertical"
    }
  },
`

Can you share how you do it?

**Reply by Gyang53** [reply]
`  {
    "context": "(ProjectPanel && not_editing)",
    "bindings": {
      "alt-o": "project_panel::OpenPermanent"
    }
  },
  {
    "context": "Pane",
    "bindings": {
      "alt-l": "pane::SplitRight",
      "alt-h": "pane::SplitLeft",
      "alt-k": "pane::SplitUp",
      "alt-j": "pane::SplitDown"
    }
`
You can refer to my implementation method. Press alt+o to open it. After opening, split the screen up, down, left and right as needed.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
